### PR TITLE
perf: :zap: add an index to the username field in data sharing consen…

### PR DESCRIPTION
**Pull request by edx-abolger**
…t so queries using the username don't take so long

To be used when running update username scripts and other username based lookups.

**Merge checklist:**
- [ N/A ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ N/A  ] `make static` has been run to update webpack bundling if any static content was updated
- [ x ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by binodpant** [comment]
This seems fine to me! If we query a lot based off of username, this makes total sense. I have not added an index in edx models before but looks like this is fully hands-off once done, based on https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations#EverythingAboutDatabaseMigrations-Howtoaddindextoexistingtable(AWSAurora)

**Review by binodpant — APPROVED** [review]
lgtm

File: consent/migrations/0005_auto_20210413_1644.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('consent', '0004_datasharingconsenttextoverrides'),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name='datasharingconsent',
+            index=models.Index(fields=['username'], name='consent_dat_usernam_fae23a_idx'),
```
Comment (raq929) [inline consent/migrations/0005_auto_20210413_1644.py:15]: What's behind this name for it?

Comment (edx-abolger) [reply consent/migrations/0005_auto_20210413_1644.py:15]: That's randomly generated by the makemigration tool.

**Review by iloveagent57 — APPROVED** [review]
Looks good, needs boilerplate.

**Comment by edx-abolger** [comment]
Not going to merge this, the plan was to merge this, run the job that this sped up, and then possibly remove the index, but the job has now been completed without this enhancement, and future related items should be designed as a group.

**Pull request closed without merging**